bdikit/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/cl_api.py

```diff
@@ -105,15 +114,58 @@ def _sample_to_15_rows(self, table: pd.DataFrame):
                 table = unique_rows.sample(n=15, random_state=1)
         return table
 
+    def _check_gdc_cache(self, table: pd.DataFrame):
+
+        gdc_df = pd.read_csv(GDC_TABLE_PATH)
+        gdc_hash = hash_dataframe(gdc_df)
+
+        table_hash = hash_dataframe(table)
```

Comment:
If we store `gdc_df` globally, would it work if we check if the `table` and `gdc_df` are the same object reference to avoid hashing `table` when it is not necessary?

For example, everywhere in the library that needs the file `GDC_TABLE_PATH` could use a function from `utils.py` that loads it (if it hasn't been loaded yet, otherwise we just return a previously loaded DataFrame). Then, here `table` and `gdc_df` would likely point to the same reference.

Reply:
As the cost of hashing table is low, I would rather test for the hash values. It would be safer than relying on object reference values. I guess a middle ground would be testing something like the number/names of attributes before comparing the hash. I have included the function to load and return the (global) GDC. We can use it everywhere GDC is accessed, and it will only be loaded once.

Reply:
@aecio  @roquelopez, please let me know if you have further iterations for this PR.

Reply:
Sounds good to me, I'm marking this as resolved.